systemtest/gencorpora/catbulk.go

```diff
@@ -0,0 +1,147 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package gencorpora
+
+import (
+	"bufio"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v8/esutil"
+)
+
+// CatBulkServer wraps http server and a listener to listen
+// for ES requests on any available port
+type CatBulkServer struct {
+	listener net.Listener
+	server   *http.Server
+	Addr     string
+
+	writer io.WriteCloser
+}
+
+// GetCatBulkServer returns a HTTP Server which can serve as a
+// fake ES server writing the response of the bulk request to the
+// provided writer. Writes to the provided writer must be thread safe.
+func GetCatBulkServer() (*CatBulkServer, error) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return nil, err
+	}
+
+	writer, err := getWriter(gencorporaConfig.WritePath)
+	if err != nil {
+		return nil, err
+	}
+
+	addr := listener.Addr().String()
+	return &CatBulkServer{
+		listener: listener,
+		Addr:     addr,
+		server: &http.Server{
+			Addr:    addr,
+			Handler: handleReq(writer),
+		},
+		writer: writer,
+	}, nil
+}
+
+// Start starts the fake ES server on a listener.
+func (s *CatBulkServer) Start() error {
```

Comment:
Nit: maybe call this Serve? When I see "Start" I think of non-blocking methods like `os/exec.Command.Start`